dbms/src/Storages/MergeTree/DiskSpaceMonitor.cpp

```diff
@@ -1,10 +1,199 @@
 #include <Storages/MergeTree/DiskSpaceMonitor.h>
 
+#include <Common/escapeForFileName.h>
+
 namespace DB
 {
 
-UInt64 DiskSpaceMonitor::reserved_bytes;
-UInt64 DiskSpaceMonitor::reservation_count;
+std::map<String, DiskSpaceMonitor::DiskReserve> DiskSpaceMonitor::reserved;
 std::mutex DiskSpaceMonitor::mutex;
 
+DiskSelector::DiskSelector(const Poco::Util::AbstractConfiguration & config, const std::string & config_prefix) {
+    Poco::Util::AbstractConfiguration::Keys keys;
+    config.keys(config_prefix, keys);
+
+    constexpr auto default_disk_name = "default";
+    for (const auto & disk_name : keys)
+    {
+        UInt64 keep_free_space_bytes = config.getUInt64(config_prefix + "." + disk_name + ".keep_free_space_bytes", 0);
+        String path;
+        if (config.has(config_prefix + "." + disk_name + ".path")) {
+            path = config.getString(config_prefix + "." + disk_name + ".path");
+        }
+
+        if (disk_name == default_disk_name) {
+            if (!path.empty()) {
+                ///@TODO_IGR ASK Rename Default disk to smth? ClickHouse disk? DB disk?
```

Comment:
There are "default" disk and "default" schema.
It is fine name for schema.
What abut disk?